modules/deployment/data.tf

```diff
@@ -0,0 +1,25 @@
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+
+data "aws_ecr_repository" "this" {
+  name = var.ecr_repository_name
+}
+
+data "aws_iam_role" "code_build" {
+  count = var.enabled && var.code_build_role != "" ? 1 : 0
+  name  = var.code_build_role
+}
+
+data "aws_iam_role" "code_pipeline" {
+  count = var.enabled && var.code_pipeline_role != "" ? 1 : 0
+  name  = var.code_pipeline_role
+}
+
+locals {
+  iam_path = "/ecs/deployment/"
+
+  create_code_build_iam    = var.enabled && var.code_build_role == ""
```

Comment:
this is used only in `iam_code_build.tf` maybe we should put it there for readability